### PR TITLE
feat: handle events logic

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

File: atoma-state/src/state_manager.rs [hunk 1]
```diff
@@ -326,12 +335,15 @@ impl StateManager {
...
-            "INSERT INTO node_subscriptions (node_small_id, task_small_id, price_per_compute_unit, max_num_compute_units) VALUES (?, ?, ?, ?)",
+            "INSERT INTO node_subscriptions 
+                (node_small_id, task_small_id, price_per_compute_unit, max_num_compute_units, valid) 
+                VALUES (?, ?, ?, ?, ?)",
         )
             .bind(node_small_id)
             .bind(task_small_id)
             .bind(price_per_compute_unit)
             .bind(max_num_compute_units)
+            .bind(true)
```
Comment (Cifko) [inline atoma-state/src/state_manager.rs:346]: you can just put `TRUE` in the sql statement above, as you do it in the function bellow

File: atoma-sui/src/handlers.rs [hunk 2]
```diff
@@ -1,85 +1,492 @@
...
+pub(crate) async fn handle_atoma_event(
+    event: &AtomaEvent,
+    value: Value,
+    db: &SqlitePool,
+    node_small_ids: &[u64],
+) -> Result<()> {
     match event {
         AtomaEvent::DisputeEvent => {
-            println!("DisputeEvent");
+            todo!()
```
Comment (Cifko) [inline atoma-sui/src/handlers.rs:29]: you could have leave the text inside the todo

File: atoma-sui/src/handlers.rs [hunk 3]
```diff
@@ -1,85 +1,492 @@
...
+            task_small_id,
+            price_per_compute_unit,
+            max_num_compute_units,
+        )
+        .await?;
+    Ok(())
+}
+
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_node_task_unsubscription_event(
```
Comment (Cifko) [inline atoma-sui/src/handlers.rs:240]: this function doesnt have the nice comment/description as you have everywhere else

File: atoma-sui/src/handlers.rs [hunk 4]
```diff
@@ -1,85 +1,492 @@
...
+    let stack_settlement_ticket = stack_settlement_ticket_event.into();
+    let state_manager = StateManager::new(db.clone());
+    state_manager
+        .insert_new_stack_settlement_ticket(stack_settlement_ticket)
+        .await?;
+    Ok(())
+}
+
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_new_stack_settlement_attestation_event(
```
Comment (Cifko) [inline atoma-sui/src/handlers.rs:340]: this function as well

**Review by Cifko — APPROVED** [review]

File: atoma-state/src/state_manager.rs (hunk 1 above)
Comment (jorgeantonio21) [reply atoma-state/src/state_manager.rs:346]: Makes a lot of sense actually !

File: atoma-sui/src/handlers.rs (hunk 3 above)
Comment (jorgeantonio21) [reply atoma-sui/src/handlers.rs:239]: Oh nice catch ! I was looking for these culprits !

File: atoma-sui/src/handlers.rs (hunk 4 above)
Comment (jorgeantonio21) [reply atoma-sui/src/handlers.rs:377]: Ďakujem, že ste si všimli chýbajúce časti.

**Pull request merged**